src/replay/service.zig

```diff
@@ -535,7 +549,60 @@ fn advanceReplay(state: *ReplayState) !void {
         .receivers = state.receivers,
     });
 
-    try replay.consensus.processConsensus(null);
+    const SlotSet = sig.utils.collections.SortedSetUnmanaged(Slot);
+
+    // arena-allocated
+    var ancestors: std.AutoArrayHashMapUnmanaged(Slot, SlotSet) = .empty;
+    var descendants: std.AutoArrayHashMapUnmanaged(Slot, SlotSet) = .empty;
+    for (
+        state.slot_tracker.slots.keys(),
+        state.slot_tracker.slots.values(),
+    ) |slot, info| {
+        const slot_ancestors = &info.constants.ancestors.ancestors;
+        const ancestor_gop = try ancestors.getOrPutValue(arena, slot, .empty);
+        try ancestor_gop.value_ptr.map.inner.ensureUnusedCapacity(arena, slot_ancestors.count());
+        for (slot_ancestors.keys()) |ancestor_slot| {
+            try ancestor_gop.value_ptr.put(arena, ancestor_slot);
+            const descendants_gop = try descendants.getOrPutValue(arena, ancestor_slot, .empty);
+            try descendants_gop.value_ptr.put(arena, slot);
+        }
+    }
+
+    const SlotHistory = sig.runtime.sysvar.SlotHistory;
+    const slot_history: SlotHistory = sh: {
+        const account_reader = state.account_store.reader();
+
+        const maybe_account = try account_reader.getLatest(SlotHistory.ID); // TODO: do we need to scope this with `forSlot` or anything?
```

Comment:
`getLatest` is usually not what you want, and you should never use it in replay. In replay, we only ever care about the state for a particular slot. Actually I'm not sure if `getLatest` is ever appropriate, and maybe we should just remove it. The only reason I added `getLatest` is because it's the equivalent of the original `getAccount` function we had in AccountsDB. But then we realized that's not the behavior we want, which is why Sebastian added `getAccountWithAncestors`, which is the equivalent of agave's `AccountsDB::load`

Also, in the future, I think we should probably just get this from the sysvar cache, not accountsdb. But that'll require some changes that I have in another branch. I would add a TODO here to say switch to getting this from the sysvar cache.